lib/utils/slices/slices.go

```diff
@@ -0,0 +1,40 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package slices
+
+import (
+	"cmp"
+	"slices"
+)
+
+// CollectValues applies a function to all elements of a slice and collects them.
+// The function returns the value to collect and whether the current element should be skipped.
+// Returned values are sorted and deduplicated.
+func CollectValues[T any, S cmp.Ordered](ts []T, fn func(T) (s S, skip bool)) []S {
```

Comment:
This function confused me until I read its definition. This looks to me like a classic map + filter except the results are sorted and unique. Can we call this function something like `FilterMapUnique` or `FilterMapUniqueSorted`? (And if you do that, I'd probably negate `skip` and call it `include` instead, since that's how I'd expect a filter function to work)